analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -205,20 +249,23 @@ private ProgramState ProcessInvocation(SymbolicContext context, IInvocationOpera
                     nonEmptyAccess.Add(context.Operation.Instance);
                 }
             }
-            if (instance.Type.DerivesOrImplementsAny(TrackedCollectionTypes))
```

Comment:
Why was this if removed? It can now trigger on any method with the same name, no matter what the type is.

Reply:
It was moved up